### PR TITLE
Fix: trim social share snippet

### LINKED ISSUE #138: Trim social share description content to closest sentence end.
When we generate the snippet of story content that gets shared on social, we should remove 'widowed' sentence fragments by trimming back to the closest period. Small enhancement, but looks a lot cleaner. 

CleanShot 2021-04-28 at 10 00 33@2x

### PR DESCRIPTION
**Pull request by LuckeeDev**
I removed the last three characters from the story snippet that we're sharing and I inserted `...` at the last of the string.

This closes #138.

### CODE REVIEW

**Comment by silent1mezzo** [comment]
@LuckeeDev Thanks for the submission. The problem wasn't exactly that we were missing the `...` (we actually have room to add that without removing the 3 other characters). The problem is that we cut off full words.

e.g. https://www.mycovidstory.ca/story/cko2x1fgw00000vl4kv7valru

```"I live alone. I'm a very sociable retired woman in my late 50's.
I miss being with my friends so much it hurts!
I gave up trying to book a shot thru t" From N5Z https://www.mycovidstory.ca/story/cko2x1fgw00000vl4kv7valru via @MyCOVIDStory_CA```

With this change, this would still be

```"I live alone. I'm a very sociable retired woman in my late 50's.
I miss being with my friends so much it hurts!
I gave up trying to book a shot thr..." From N5Z https://www.mycovidstory.ca/story/cko2x1fgw00000vl4kv7valru via @MyCOVIDStory_CA ```

I think the better solution would be to slice the content and only include full words. e.g.

```"I live alone. I'm a very sociable retired woman in my late 50's.
I miss being with my friends so much it hurts!
I gave up trying to book a shot ..." From N5Z https://www.mycovidstory.ca/story/cko2x1fgw00000vl4kv7valru via @MyCOVIDStory_CA ```

**Comment by silent1mezzo** [comment]
The failed test is because a string < 150 characters shouldn't be cut off https://github.com/my-covid-story/www/pull/149/checks?check_run_id=2468323252

**Comment by LuckeeDev** [comment]
Oh sorry, I didn't understand correctly. I'll work on it ASAP.

**Comment by LuckeeDev** [comment]
It should be fixed now! Let me know what you think @silent1mezzo :)

File: lib/social.ts [hunk 1]
```diff
@@ -1,6 +1,21 @@
...
+    const storySnippet = content.slice(0, contentSize)
+
+    // Trim the story to the last space
+    const trimmedStorySnippet = storySnippet.substring(
+      0,
+      Math.min(storySnippet.length, storySnippet.lastIndexOf(' '))
+    )
+
+    return `"${trimmedStorySnippet}..." ${storyCite(story).slice(0, citeSize)}`
+  }
```
Comment (silent1mezzo) [inline lib/social.ts:20]: One final nitpick, the else isn't necessary in this case. Removing it makes it slightly easier to read.

```suggestion
  if (content.length < contentSize) {
    return `"${content}" ${cite}`
  }
  const storySnippet = content.slice(0, contentSize)

  // Trim the story to the last space
  const trimmedStorySnippet = storySnippet.substring(
    0,
    Math.min(storySnippet.length, storySnippet.lastIndexOf(' '))
  )

  return `"${trimmedStorySnippet}..." ${storyCite(story).slice(0, citeSize)}`
```

**Review by silent1mezzo — APPROVED** [review]
One final suggestion to clean up the code but otherwise looks good! This is great!

**Comment by LuckeeDev** [comment]
Good catch @silent1mezzo!

**Comment by silent1mezzo** [comment]
Awesome! Amazing work @LuckeeDev

**Comment by LuckeeDev** [comment]
It's a pleasure!

**Pull request merged**